groovy-language-server/src/main/java/com/palantir/ls/groovy/GroovycWrapper.java

```diff
@@ -188,11 +272,17 @@ private Pattern getQueryPattern(String query) {
     }
 
     private void addAllSourcesToCompilationUnit() {
+        // Figure out which files will be replaced and therefore need to be excluded
+        Set<Path> filesToExclude = originalSourceToChangedSource.keySet();
```

Comment:
looks like filesToExclude is only used at one place, maybe we can directly change
`if (!filesToExclude.contains(file.toPath()) && file.isFile() && Files.getFileExtension(file.getAbsolutePath()).equals(GROOVY_EXTENSION))`
to 
`if (!originalSourceToChangedSource.containsKey(file.toPath()) && file.isFile() && Files.getFileExtension(file.getAbsolutePath()).equals(GROOVY_EXTENSION))`